tests/test_frame.py

```diff
@@ -1342,7 +1342,8 @@ def test_getmultiindex_columns() -> None:
 
 def test_frame_getitem_isin() -> None:
     df = pd.DataFrame({"x": [1, 2, 3, 4, 5]}, index=[1, 2, 3, 4, 5])
-    check(assert_type(df[df.index.isin([1, 3, 5])], pd.DataFrame), pd.DataFrame)
+    bool_array = df.index.isin([1, 3, 5])
+    check(assert_type(df[bool_array], pd.DataFrame), pd.DataFrame)
```

Comment:
The type checkers think this is a `pd.Series`, but I can't see how that can be the case. I tried the `test_not_hashable` approach on `np_ndarray_bool` and it did not fail (so presumably it is not a `Hashable`. Any ideas?

Reply:
Are you sure?  I just did this with the current version of the stubs:
```python
import pandas as pd

def test_frame_getitem_isin() -> None:
    df = pd.DataFrame({"x": [1, 2, 3, 4, 5]}, index=[1, 2, 3, 4, 5])
    bool_array = df.index.isin([1, 3, 5])
    sdf = df[bool_array]
    reveal_type(sdf)
```
Both `mypy` and `pyright` said that `sdf` was `DataFrame`

Reply:
That is after my change in this PR which updates `__getitem__`